### ISSUE TITLE
UpdateCountComments/UpdateCountDiscussions methods naming is confusing

### ISSUE THREAD

**Issue by R-J** [vanilla/vanilla#4082]
There is
commentModel->updateCommentCount => updates GDN_Discussion
commentModel->updateUserCommentCount**s** => updates GDN_UserDiscussion
commentModel->updateUser => updates GDN_User
discussionModel->updateDiscussionCount => updates GDN_Category
discussionModel->updateUserDiscussionCount => updates GDN_User

Sometimes the **table** that is altered is **not named at all**, **sometimes** it is mentioned **complete** and **sometimes** only **partly**.
The **column** that is changed is **never** mentioned **correct** and **not** even **consistent**ly used wrong. The column name is not CommentCount(s)/DIscussionCount, but CountComments/CountDiscussions.
You could only know what they are doing if you see all of them or read their documentation (the last mentioned Not all of them are documented 

I would recommend to rename them all to actionTable(s)Column. Not sure about that "s" so I just enclose it in brackets...

commentModel->updateDiscussion(s)CountComments => updates GDN_Discussion
commentModel->updateUserDiscussion(s)CountComments => updates GDN_UserDiscussion
commentModel->updateUser(s)CountComments => updates GDN_User
discussionModel->updateCategory(s)CountDiscussions => updates GDN_Category
discussionModel->updateUser(s)CountDiscussions=> updates GDN_User

Only using updateTablename would be okay in some cases:

commentModel->updateUserDiscussion => updates GDN_UserDiscussion
commentModel->updateUser => updates GDN_User
discussionModel->updateUser => updates GDN_User

But now for discussionModel->updateCategory and commentModel->updateDiscussion since there is more than one discussion/comment specific column in this tables. So to keep method names short it could be done like that:

commentModel->updateUserDiscussion => updates GDN_UserDiscussion
commentModel->updateUser => updates GDN_User
discussionModel->updateUser => updates GDN_User
discussionModel->updateCategoryCountDiscussions => updates GDN_Category
commentModel->updateDiscussionCountComments => updates GDN_Discussion

But I would really favour:
commentModel->updateDiscussionCountComments => updates GDN_Discussion
commentModel->updateUserDiscussionCountComments => updates GDN_UserDiscussion
commentModel->updateUserCountComments => updates GDN_User
discussionModel->updateCategoryCountDiscussions => updates GDN_Category
discussionModel->updateUserCountDiscussions=> updates GDN_User

This would be consistent and selfexplanatory.

**Comment by DaazKu** [comment]
I really like:

```
commentModel->updateUserDiscussion => updates GDN_UserDiscussion
commentModel->updateUser => updates GDN_User
discussionModel->updateUser => updates GDN_User
discussionModel->updateCategoryCount => updates GDN_Category
commentModel->updateDiscussionCount => updates GDN_Discussion
```

Since you already are on a model you do not need to specify that you are updating the "modelName" count. Just the target should be fine.
With the updated documentation the pattern become quite clear.

**Comment by R-J** [comment]
I think the naming for those three is obvious:
~~~
commentModel->updateUser => updates GDN_User
discussionModel->updateUser => updates GDN_User
commentModel->updateUserDiscussion => updates GDN_UserDiscussion
~~~
But I guess it should be considered that if anything else than CountDiscussions/CountComments must be updated, it either should be done in this methods or it will become somewhat strange again.


I don't think this names are self explanatory and they are not even logical:
~~~
discussionModel->updateCategoryCount => updates GDN_Category
commentModel->updateDiscussionCount => updates GDN_Discussion
~~~
They are expressing "action, table, Count*Something*". If it would be common practice that if *Something* isn't obvious, the "models type" is meant, it would be okay.

But I guess the biggest problem is  that there is no common naming practice. Either there should be one (which would be great, but also complicated and a lot of work) or method names must be self explanatory.

Just take a look at discussionModels bookmark functions. They are a mixture of everything:
~~~
getBookmarkUsers($DiscussionID) => Gets all users who have bookmarked the specified discussion.
userBookmarkCount($UserID) => Gets number of bookmarks specified user has.
bookmarkCount($DiscussionID) => Gets number of bookmarks specified discussion has (all users).

setUserBookmarkCount($UserID) => Update and get bookmark count for the specified user.

bookmarkDiscussion($DiscussionID, $UserID, &$Discussion = null) => Bookmarks (or unbookmarks) a discussion for     specified user.
bookmark          ($DiscussionID, $UserID, $Bookmarked = null)  => Bookmarks (or unbookmarks) a discussion for the specified user.
~~~
I've added some spaces to the last two in order to show another naming issue I've just stumbled upon.
<hr>
Before I got lost in details, I want to sum up that I came to the opinion that there should be a naming convention, otherwise renaming only a few methods is as good as leaving them like they are.
If names are just created based on best guess, today some people might find the current or new name brilliant and tomorrow some other guys (or even the same) wouldn't be satisfied with that name.
